File: BOJ/Stack/BJ10773.js
const fs = require('fs');
const input = fs.readFileSync('/dev/stdin')
  .toString()
  .trim()
  .split('\n')
  .map(Number);

const K = input.shift();
const stack =[];

for(const num of input) {
  num === 0 
    ? stack.pop()
    : stack.push(num);
}

let sum = 0;
while(stack.length > 0) sum += stack.pop();

console.log(sum);

File: BOJ/Stack/BJ10773.cpp
#include <iostream>
#include <stack>
using namespace std;

int K;
int a;
int sum;
stack<int> s;

int main() {
  ios::sync_with_stdio(0);
  cin.tie(0);

  cin >> K; 
  for(int i=0; i<K; i++) {
    cin >> a;
    
    if(a == 0) s.pop();
    else s.push(a);  
  }

  while(!s.empty()) {
    sum += s.top();
    s.pop();
  }
  cout << sum << '\n';
  return 0;
}